plugin/core/registry.py

```diff
@@ -86,11 +86,15 @@ def get_position(view: sublime.View, event: Optional[dict] = None) -> int:
 class LspTextCommand(sublime_plugin.TextCommand):
 
     capability = ''
+    session_name = ''
```

Comment:
Since you want to make it a public API, can we add comments here explaining the relationship between those two (`capability` wins if set, `session_name` only checked if `capability` is not set)?

Or maybe it should rather work so that both are checked always (when set)? So that we can restrict the command to both at the same time?

Reply:
> Or maybe it should rather work so that both are checked always (when set)? So that we can restrict the command to both at the same time?

Well, a helper package knows exactly which capabilities the server has, so this seems redundant.

Reply:
I guess I don't have any counter-arguments for that :)

Just that it wouldn't hurt to make things more explicit and would also make the logic more straight-forward (all respected when set rather than one having higher priority when set).